core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -197,11 +197,9 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                                                       cfg, scheme.toString(), reqTarget);
                     // Do not accept unsupported methods.
                     final HttpMethod method = headers.method();
-                    switch (method) {
-                        case CONNECT:
-                        case UNKNOWN:
-                            fail(id, headers, HttpStatus.METHOD_NOT_ALLOWED, "Unsupported method", null);
-                            return;
+                    if (method == HttpMethod.CONNECT) {
+                        fail(id, headers, HttpStatus.METHOD_NOT_ALLOWED, "Unsupported method", null);
+                        return;
```

Comment:
Can we also add a `ServerBuilder#additionalAllowedHttpMethods(String...)` and validate if the method is `UNKNOWN`?  Previous users may want to only receive standard HTTP Methods.